app/ai/voice/stt/custom_soniox.py

```
@@ -0,0 +1,425 @@
+                finished = content.get("finished")
+                if finished:
+                    await send_endpoint_transcript()
+                    logger.debug("Soniox transcription finished.")
+                    return
+
+            except json.JSONDecodeError:
+                logger.warning(f"Received non-JSON message from Soniox: {message}")
+            except Exception as e:
+                logger.warning(f"Error processing Soniox message: {e}")
+
+    async def _send_keepalive(self, silence: bytes):
```

Ruff (0.15.1)

[warning] Unused method argument: `silence`

(ARG002)